### PR TITLE
FEA-3404: export SymbolGenerator

### PR DESCRIPTION
**Pull request by matthewnitschke-wk**
There are many usages for symbol representations of dart element types. Not all of these usages fall into the category of the raw indexing and outputting of an indexed repo

This pr exports an internal class: `SymbolGenerator`, which can be used to convert dart analyzer element types in their correlating scip symbol, allowing for unique and innovative implementation for generating these scip symbols

### CODE REVIEW

**Comment by aviary3-wk** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: lib/src/symbol_generator.dart [hunk 1]
```diff
@@ -47,7 +47,7 @@ class SymbolGenerator {
     return [
       'scip-dart',
       _getPackage(element),
-      _getDescriptor(element),
+      descriptor,
```
Comment (matthewnitschke-wk) [inline lib/src/symbol_generator.dart:50]: we were unnecessary generating the descriptor twice, for every symbol (see line 43) this cleans that up

**Review by todbachman-wf — DISMISSED** [review]

**Comment by matthewnitschke-wk** [comment]
QA +1

- [x] CI does indeed pass

🚀 @Workiva/release-management-p 🚢

**Review by todbachman-wf — APPROVED** [review]

**Comment by matthewnitschke-wk** [comment]
QA +1

- [x]  CI does indeed pass

🚀 @Workiva/release-management-p 🚢

**Review by rmconsole-wf — APPROVED** [review]
+1 from RM

**Pull request merged**